File: 41-first-missing-positive/41-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& nums) {
        int n = nums.size();
        vector<int> seen(n + 1, 1);
        
        for (int num : nums){
            if (1 <= num && num <= n){
                seen[num] = 0;
            }
        }
        
        int i = 1;
        while (i < n + 1 && seen[i] == 0) {i++;}
        return i;
        }
};

File: 41-first-missing-positive/41-first-missing-positive.py
class Solution:
    def firstMissingPositive(self, nums: List[int]) -> int:
        seen = [1 for i in range(len(nums) + 2)]
        
        for num in nums: 
            if 0 <= num <= len(nums):
                seen[num] = -1
        
        for i in range(1, len(seen)):
            if seen[i] >= 0:
                return i